lib/src/createThemedComponent.js

```diff
@@ -41,32 +40,38 @@ export default function createThemedComponent(C, themedProps = []) {
       this.state = {
         theme,
         props: themedProps.reduce((res, key) => {
-          res[key] = theme.getProp(props[key]);
+          res[key] = this.getThemedPropValue(props[key], theme)
           return res;
         }, {}),
       };
     }
 
+    getThemedPropValue = (prop, theme) => {
+      if (typeof prop == 'object') {
+        return theme.getStyle(prop);
+      } else {
+        return theme.getProp(prop);
+      }
+    }
+
     // eslint-disable-next-line camelcase, react/sort-comp
     UNSAFE_componentWillMount() {
       this.unregister = registerComponent(this);
     }
-
     // eslint-disable-next-line camelcase, react/sort-comp
     UNSAFE_componentWillReceiveProps(nextProps) {
       const { theme, props } = this.state;
-
       let changed = false;
       // See if props have changed, only in which case change the state
       const newPropsState = themedProps.reduce((res, key) => {
-        const newValue = theme.getProp(nextProps[key]);
+        let newValue;
+        newValue = this.getThemedPropValue(nextProps[key], theme);
```

Comment:
@smsmdida if there isn't any reason to use `let` can we convert this to `const`.

Reply:
@smsmdida Also when you commit, can you also make sure the unnecessary changes due to whitespaces are removed, it just creates unnecessary git changes to look for.

Reply:
@syaau no reason to use let, mistake when i refactored my code :D